teammapper-frontend/mmp/src/map/handlers/nodes.ts

```diff
@@ -235,13 +235,20 @@ export default class Nodes {
                     if (x.parent.hidden && !x.hidden) {
                         this.updateNode('hidden', true, false, false, false, x.id)
                     }
-                    
+
                     if (!x.parent.hidden && x.hidden) {
                         this.updateNode('hidden', false, false, false, false, x.id)
                     }
                 })
             }
 
+            // Lengthy but definitive check to see if we have any hidden nodes after toggling
+            if (this.map.nodes.nodeChildren(this.selectedNode.id)?.filter(x => x.hidden).length > 0) {
+                this.selectedNode.hasHiddenChildNodes = true
```

Comment:
can't this be set directly, when the button is clicked (hide children)? Do we have to calculate this dynamically?

Reply:
As I see it, this property makes sense. So we have to new fields:

- hidden (on the descendants) 
- hiddenBranches (on the parent) 

So I would set them both upon button press?

Reply:
good point, maybe you can add this reason as a comment? that there is not only the button that triggers it and thats the reason you have to check here.

Reply:
Good idea, I've added some comments.